internal/pally/opts.go

```diff
@@ -47,6 +49,31 @@ func (o Opts) describe() *prometheus.Desc {
 	)
 }
 
+func (o Opts) labelPairs(variableLabelVals []string) []*dto.LabelPair {
+	n := len(o.ConstLabels) + len(o.VariableLabels)
+	if n == 0 {
+		return nil
+	}
+	pairs := make([]*dto.LabelPair, 0, n)
+	for k, v := range o.ConstLabels {
+		pairs = append(pairs, &dto.LabelPair{
+			Name:  proto.String(k),
+			Value: proto.String(v),
+		})
+	}
+	if len(variableLabelVals) != len(o.VariableLabels) {
```

Comment:
It's part of the Vector API; it's a bit foot-shooty, but also faster than passing maps everywhere.